File: app/src/main/java/com/codeartist/androidpractice/viewmodel/PropertyViewModelFactory.java
package com.codeartist.androidpractice.viewmodel;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.codeartist.androidpractice.repository.Repository;

public class PropertyViewModelFactory implements ViewModelProvider.Factory {
    private Repository repository;
    private String mParam;


    public PropertyViewModelFactory(Repository repository) {
        this.repository = repository;
    }


    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new PropertyViewModel(repository);
    }
}

File: app/src/main/java/com/codeartist/androidpractice/viewmodel/PropertyViewModelFactory.kt
package com.codeartist.androidpractice.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.codeartist.androidpractice.repository.PropertyRepository


class PropertyViewModelFactory(private val repository: PropertyRepository) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return PropertyViewModel(repository) as T
    }
}